app/Platform/Actions/UpsertGameCoreAchievementSetFromLegacyFlags.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Models\AchievementSet;
+use App\Models\Game;
+use App\Platform\Enums\AchievementSetType;
+use Illuminate\Support\Facades\DB;
+
+class UpsertGameCoreAchievementSetFromLegacyFlags
+{
+    public function execute(Game $game): void
+    {
+        $game->load(['achievements', 'gameAchievementSets']);
+
+        DB::transaction(function () use ($game) {
+            $hasCoreSets = $game->gameAchievementSets()->core()->count() > 0;
+            if ($hasCoreSets) {
+                $this->updateExistingCoreSetOfGame($game);
+            } else {
+                $newAchievementSet = $this->createNewAchievementSetFromGame($game);
+                $game->achievementSets()->attach($newAchievementSet->id, [
+                    'type' => AchievementSetType::CORE->value,
+                    'order_column' => 0,
+
+                    // Preserve the existing timestamps. Note that updated_at reuses the created_at
+                    // value, as none of the fields stored in game_achievement_sets would've changed
+                    // from the time of the first achievement promotion.
+                    'created_at' => $newAchievementSet->created_at,
+                    'updated_at' => $newAchievementSet->created_at,
+                ]);
```

Comment:
Should this also auto-generate a subset record for subsets?
i.e. game 25176 would have two entries in the game_achievement_sets table:
1) `game_id 25176 core` -> `achievement_set_id_N`
2) `game_id 9823 will_be_subset` -> `achievement_set_id_N`

Or is the intent to do that manually at some point, or to change the `25176 core` set to `will_be_subset`?

Reply:
The intent is to do it manually later with the `will_be_*` stuff and new tools provided for devs in the management app. They will explicitly set what kind of subset it is rather than us trying to intuit it.